internal/hcsoci/network.go

```diff
@@ -68,3 +70,63 @@ func GetNamespaceEndpoints(ctx context.Context, netNS string) ([]*hns.HNSEndpoin
 	}
 	return endpoints, nil
 }
+
+// NetworkSetup is used to abstract away the details of setting up networking
+// for a container.
+type NetworkSetup interface {
```

Comment:
Can we find a better place for these types than `hcsoci`? They don't seem to have anything to do with the OCI spec.

Reply:
New `network` package?

Reply:
It seems specific to configuring the UVM's networking, so maybe just in the UVM package?

Reply:
Sure

Reply:
Crud there's an import cycle because we call hcsoci.GetNamespaceEndpoints in the local case, and hcsoci imports internal/uvm

Reply:
https://github.com/microsoft/hcsshim/blob/251b969e31699d4b72750d5855dc3403e85a300a/internal/hcsoci/network.go#L50 Should we just make this a function in the `hcn` package?

Reply:
Although we use the hns definitions for network setup so either or

Reply:
I moved this to the UVM package